posts/2024-02-19-waku-network-tech-overview.mdx

```diff
@@ -0,0 +1,172 @@
+		
+Hence, there is a risk of someone pushing gigabits of data to the network. This can be a problem at different levels:
+		
+- Bandwidth usage: This may lead to a user's bandwidth being hogged and affecting other services (streaming, staking) or a surprise bill.
+- Connectivity: if a node does not have enough bandwidth to send/receive all messages on the gossipsub layer, then their behaviour may be seen as incorrect by other nodes that may disconnect from it.
+- Reliability: If the traffic is higher than the available bandwidth, then a node may not be able to send and receive messages reliably.
+- Other resources: memory usage is correlated with the traffic, and so is disk space for a store service.
+		
+So rather than defining what a "spam" message looks like, we introduced rate limit on the network to enable a fair use of Waku, with a capping of the bandwidth usage per shard.
+		
+This is done with RLN, or Rate Limiting Nullifier, which limits the rate of messages a given publisher sends.
+It is currently set to 1 msg/s. Coupled with a maximum message size (150kB) and a maximum number of publishers (80k, TBD), we can assume a maximum bandwidth usage per shard (around 10Mbps).
+		
+Rate-limiting publishers in a censorship-resistant and private manner is difficult; this is why we are using zero-knowledge technology:
+		
+1. Users push their RLN credentials to a smart contract (currently on Ethereum Sepolia testnet).
+2. Nodes keep track of all registered memberships on the contract.
+3. When sending a message, the user attaches an RLN proof to the message with the current epoch (timestamp in seconds).
+4. Nodes can verify the proof without being able to correlate the user's Ethereum address (used on the smart contract) with the message being sent, preserving anonymity.
+5. If a user attempts to send more than 1 msg/s, nodes can detect this and drop the message surplus or spam, protecting the network.
+
+### Servicing Mostly-offline and Resource-restricted Devices
+Finally, how is Waku useful to resource-restricted devices, such as mobile phones and web browsers?
+Waku defines a number of [request-response protocols (https://rfc.vac.dev/spec/10/#requestreply-domain) to enable such devices to access the Waku network without having to be always online or consume extensive amounts of bandwidth, i.e., without participating in the Waku Relay network.
+The light push protocol (links to docs) enables a _light client_ to send a message to be forwarded to the Waku Relay network, with acknowledgement of reception from the remote node.
+		
+The filter protocol enables a _light client_ to subscribe to a remote peer and only request a subset of messages instead of all messages transmitted on a shard.
```

Comment:
```suggestion
The [filter protocol](https://docs.waku.org/learn/concepts/protocols/#filter) enables a _light client_ to subscribe to a remote peer and only request a subset of messages instead of all messages transmitted on a shard.
```